src/main/java/co/cask/http/ExceptionHandler.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.http;
+
+import org.jboss.netty.handler.codec.http.HttpRequest;
+
+/**
+ * Handles exceptions and provides a response via the {@link HttpResponder}.
+ */
+public interface ExceptionHandler {
```

Comment:
make this an abstract class with the default implementation just log and response with 500